drivers/lora/sx12xx_common.c

```diff
@@ -8,18 +8,23 @@
 #include <drivers/gpio.h>
 #include <drivers/lora.h>
 #include <logging/log.h>
+#include <sys/atomic_builtin.h>
 #include <zephyr.h>
 
 /* LoRaMac-node specific includes */
 #include <radio.h>
 
 #include "sx12xx_common.h"
 
+#define MODEM_ACQUIRE(data) atomic_cas(&data.modem_usage, 0, 1)
+#define MODEM_RELEASE(data) atomic_cas(&data.modem_usage, 1, 0)
```

Comment:
Personally I would prefer a static function that operates on atomic variable and calls `Radio.Sleep()` implicitly. For example:
```
static inline void modem_release(struct sx12xx_data *data)
{
	if (atomic_cas(&data->modem_usage, 1, 0)) {
		Radio.Sleep();
	}	
}
```
So instead of:
```
	if (MODEM_RELEASE(dev_data)) {
		Radio.Sleep();
	}
```
we could have:
```
	modem_release(dev_data);
```

Reply:
We would still have the current pattern for the acquire though, as that needs to return.
My preference is to keep both patterns the same.

Reply:
My way of thinking is that "acquire" can fail if modem is busy, but "release" is an always successful operation. Code that handles LoRa send() and recv() should be only interested in releasing the modem whenever it is no longer used. Power management (calling Radio.Sleep()) is something to be handled in single place, so within `modem_acquire()` and `modem_release()`.

Reply:
One thing that I noticed now is that with this commit we indeed protect from double `Radio.Sleep()`. But there is also a possible case with:
```
	if (MODEM_RELEASE(dev_data)) {
		Radio.Sleep();
	}
```
that `MODEM_RELEASE()` can be already called and another thread can start `lora_send()` or `lora_recv()` before actually `Radio.Sleep()` will be called. So there is a (not likely, but still) possible race condition here.

Reply:
Good point. I didn't spot that particular race. Being unlikely just means it will happen in the field when it's running on a sensor node that has been nailed to a tree top.